libdnf5/base/goal.cpp

```diff
@@ -2252,11 +2396,21 @@ base::Transaction Goal::resolve() {
             libdnf5::Logger::Level::WARNING);
     }
 
+    //TODO(amatej): Add conditional check that no extra packages were added by the solver
+
     transaction.p_impl->set_transaction(p_impl->rpm_goal, module_sack, ret);
 
     return transaction;
 }
 
+
+void Goal::add_transaction_replay(
+    const std::string & json_serialized_transaction, libdnf5::GoalJobSettings & settings) {
+    //TODO(amatej): allow only one replay per goal?
```

Comment:
I guess that we should add a limitation (exception)  or support more transactions. The current implementation performs the last provided transaction, and I don't think that this is right.
If we will add support only for one file per goal we can anytime extend the support to multi transactions support without changing API, ABI, or introducing a new exception. If we will start with a support of multiple files then adding the restriction might be painful. May I ask you for your preferences?

Reply:
I think limiting to one is the right approach. You are right we can always extend it later if there is a use case.